streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -575,6 +577,8 @@ class StreamingContext private[streaming] (
           try {
             validate()
 
+            registerProgressListener()
```

Comment:
I understand you are suggesting that the unregister is important, but do we need this? This seems to be a behavior change.

Reply:
I think so. I believe``StreamingTab`` shouldn't be responsible for registering/unregistering the listener as it could be and even already used in other place (metrics). Moreover seems there is also a bug that if ui is disabled, listener isn't registered and metrics aren't reported.

Reply:
Yes I tend to agree. The behavior change is registering listeners when when the UI is disabled, but that seems like possibly a fix (we'd have to look deeper to figure out whether something _else_ registers it in this case, like `StreamingSource`). This seems like a better place to manage it, certainly unregistering, and if it's unregistered here it should be registered here.